### PR TITLE
chore(docs): Add docs for custom pydantic types with __get_validators__

### PR DESCRIPTION
**Pull request by thejaminator**
Adds documentation for users to convert pydantic models that use fields that are a custom type.

[Discord thread](https://discord.com/channels/689806334337482765/932705359334899883/932817103835254815)

Not sure where to place it. I put it at the bottom of docs. 
For a follow up MR. [We should catch / detect this error beforehand and suggest what to do?](https://discord.com/channels/689806334337482765/932705359334899883/932746286334087168)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/integrations/pydantic.md [hunk 1]
```diff
@@ -264,3 +264,52 @@ input_data = UserInput(id='abc', name='Jake')
 # this will run pydantic's validation
 instance = input_data.to_pydantic()
 ```
+
+### Classes with \_\_get_validators\_\_
+
+Pydantic BaseModels may define a custom type with [\_\_get_validators\_\_](https://pydantic-docs.helpmanual.io/usage/types/#classes-with-__get_validators__)
+logic. You will need to add a scalar type and add the mapping to the `scalar_overrides`
```
Comment (thejaminator) [inline docs/integrations/pydantic.md:271]: idk what to call it. pydantic calls it "Classes with __get__validators__" so i'm just going to call it that

File: docs/integrations/pydantic.md [hunk 2]
```diff
@@ -264,3 +264,52 @@ input_data = UserInput(id='abc', name='Jake')
 # this will run pydantic's validation
 instance = input_data.to_pydantic()
 ```
+
+### Classes with \_\_get_validators\_\_
```
Comment (patrick91) [inline docs/integrations/pydantic.md:268]:
```suggestion
### Classes with `\_\_get_validators\_\_`
```

File: docs/integrations/pydantic.md [hunk 3]
```diff
@@ -264,3 +264,52 @@ input_data = UserInput(id='abc', name='Jake')
...
+
+Pydantic BaseModels may define a custom type with [\_\_get_validators\_\_](https://pydantic-docs.helpmanual.io/usage/types/#classes-with-__get_validators__)
```
Comment (patrick91) [inline docs/integrations/pydantic.md:270]:
```suggestion
Pydantic BaseModels may define a custom type with [`\_\_get_validators\_\_`](https://pydantic-docs.helpmanual.io/usage/types/#classes-with-__get_validators__)
```

**Review by patrick91 — APPROVED** [review]
Looks good!

File: docs/integrations/pydantic.md [hunk 4]
```diff
@@ -264,3 +264,52 @@ input_data = UserInput(id='abc', name='Jake')
 # this will run pydantic's validation
 instance = input_data.to_pydantic()
 ```
+
+### Classes with `\_\_get_validators\_\_`
```
Comment (patrick91) [inline docs/integrations/pydantic.md:268]:
```suggestion
### Classes with `__get_validators__`
```

File: docs/integrations/pydantic.md [hunk 5]
```diff
@@ -264,3 +264,52 @@ input_data = UserInput(id='abc', name='Jake')
...
+
+Pydantic BaseModels may define a custom type with [`\_\_get_validators\_\_`](https://pydantic-docs.helpmanual.io/usage/types/#classes-with-__get_validators__)
```
Comment (patrick91) [inline docs/integrations/pydantic.md:270]:
```suggestion
Pydantic BaseModels may define a custom type with [`__get_validators__`](https://pydantic-docs.helpmanual.io/usage/types/#classes-with-__get_validators__)
```

**Pull request merged**